driver/service.go

```diff
@@ -241,13 +243,24 @@ func (u *upCloudService) resizeBlockDevice(ctx context.Context, uuid string, new
 }
 
 func (u *upCloudService) createStorageBackup(ctx context.Context, uuid, title string) (*upcloud.StorageDetails, error) {
+	// check that a backup creation is not currently in progress
+	storage, err := u.getStorageByUUID(ctx, uuid)
+	if err != nil {
+		return nil, err
+	}
+
+	if storage.State == upcloud.StorageStateBackuping {
+		return nil, status.Errorf(codes.Aborted, "cannot create snapshot for volume with backup in progress")
```

Comment:
Maybe we could define custom error (top of the file), something like:
```go
var errUpCloudBackupInProgress = errors.New("upcloud: backup creation is currently in progress")
```
And return `errUpCloudBackupInProgress` here so that we could return `codes.Aborted` in `controller.go` .